StripeFinancialConnections/StripeFinancialConnections/Source/Native/InstitutionPicker/InstitutionSearchBar.swift

```diff
@@ -139,6 +150,26 @@ final class InstitutionSearchBar: UIView {
         layer.borderColor = searchBarBorderColor.cgColor
         layer.borderWidth = searchBarBorderWidth
     }
+
+    private var searchIndicatorActive: Bool = false
+
+    func updateSearchingIndicator(_ isSearching: Bool) {
+        guard isSearching != self.searchIndicatorActive else { return }
```

Comment:
I believe this is so we don't add the animation to the layer multiple times? 

If correct, maybe one way to make this explicit & remove the extra variable would be to do something like:

```

func animateSearchIndicator(_ shouldAnimateSearchIndicator: Bool) {
        if shouldAnimateSearchIndicator {
            let isAnimationActive = textField.layer.animation(forKey: "pulseAnimation") != nil
            if !isAnimationActive {
                let opacityAnimation = CABasicAnimation(keyPath: "opacity")
                opacityAnimation.fromValue = 0.6
                opacityAnimation.toValue = 0.25
                opacityAnimation.repeatCount = 10000
                opacityAnimation.duration = 0.3
                opacityAnimation.autoreverses = true
    
                searchIconView.layer.add(opacityAnimation, forKey: "pulseAnimation")
            }
        } else {
            searchIconView.layer.removeAnimation(forKey: "pulseAnimation")
        }
}
```